components/eamxx/src/diagnostics/tests/aerocom_cld_test.cpp

```diff
@@ -0,0 +1,303 @@
+                         grid->name());
+  FieldIdentifier cd_fid("cldfrac_tot", scalar2d_layout, nondim, grid->name());
+  FieldIdentifier nc_fid("nc", scalar2d_layout, kg / kg, grid->name());
+
+  Field tm(tm_fid);
+  tm.allocate_view();
+  tm.get_header().get_tracking().update_time_stamp(t0);
+  Field pd(pd_fid);
+  pd.allocate_view();
+  pd.get_header().get_tracking().update_time_stamp(t0);
+  Field pm(pm_fid);
+  pm.allocate_view();
+  pm.get_header().get_tracking().update_time_stamp(t0);
+  Field qv(qv_fid);
+  qv.allocate_view();
+  qv.get_header().get_tracking().update_time_stamp(t0);
+  Field qc(qc_fid);
+  qc.allocate_view();
+  qc.get_header().get_tracking().update_time_stamp(t0);
+  Field qi(qi_fid);
+  qi.allocate_view();
+  qi.get_header().get_tracking().update_time_stamp(t0);
+  Field ec(ec_fid);
+  ec.allocate_view();
+  ec.get_header().get_tracking().update_time_stamp(t0);
+  Field ei(ei_fid);
+  ei.allocate_view();
+  ei.get_header().get_tracking().update_time_stamp(t0);
+  Field cd(cd_fid);
+  cd.allocate_view();
+  cd.get_header().get_tracking().update_time_stamp(t0);
+  Field nc(nc_fid);
+  nc.allocate_view();
+  nc.get_header().get_tracking().update_time_stamp(t0);
+
+  // Construct random number generator stuff
+  using RPDF = std::uniform_real_distribution<Real>;
+  RPDF pdf(0, 0.05);
+  auto engine = scream::setup_random_test();
+
+  // Construct the Diagnostics
+  std::map<std::string, std::shared_ptr<AtmosphereDiagnostic>> diags;
+  auto &diag_factory = AtmosphereDiagnosticFactory::instance();
+  register_diagnostics();
+
+  ekat::ParameterList params;
+
+  REQUIRE_THROWS(
+      diag_factory.create("AeroComCld", comm, params));  // No 'AeroComCld Kind'
+  params.set<std::string>("AeroComCld Kind", "Foo");
+  REQUIRE_THROWS(diag_factory.create("AeroComCld", comm,
+                                     params));  // Invalid 'AeroComCld Kind'
+
+  constexpr int ntests = 3;
+  for(int itest = 0; itest < ntests; ++itest) {
+    // Randomize everything to add ensure resiliency
+    randomize(tm, engine, pdf);
+    randomize(pd, engine, pdf);
+    randomize(pm, engine, pdf);
+    randomize(qv, engine, pdf);
+    randomize(qc, engine, pdf);
+    randomize(qi, engine, pdf);
+    randomize(ec, engine, pdf);
+    randomize(ei, engine, pdf);
+    randomize(cd, engine, pdf);
+    randomize(nc, engine, pdf);
+
+    // Create and set up the diagnostic
+    params.set<std::string>("AeroComCld Kind", "Top");
+    auto diag = diag_factory.create("AeroComCld", comm, params);
+
+    diag->set_grids(gm);
+
+    diag->set_required_field(tm);
+    diag->set_required_field(pd);
+    diag->set_required_field(pm);
+    diag->set_required_field(qv);
+    diag->set_required_field(qc);
+    diag->set_required_field(qi);
+    diag->set_required_field(ec);
+    diag->set_required_field(ei);
+    diag->set_required_field(cd);
+    diag->set_required_field(nc);
+
+    // tm.sync_to_dev();
```

Comment:
When you're ready to merge, let's also rm these commented lines.

Reply:
`randomize()` above syncs to device already.

Reply:
They aren't. The `randomize` function randomizes the host copy of the field, and then call `sync_to_host()`. So the fields are already synced.